doc/development.md

```diff
@@ -59,13 +59,23 @@ resources from our ["testing" configuration][], you can configure `nextstrain`
 with the same, e.g.:
 
     export NEXTSTRAIN_DOT_ORG=http://localhost:5000
-    export NEXTSTRAIN_COGNITO_USER_POOL_ID="$(jq -r .COGNITO_USER_POOL_ID ../nextstrain.org/env/testing/config.json)"
-    export NEXTSTRAIN_COGNITO_CLI_CLIENT_ID="$(jq -r .OAUTH2_CLI_CLIENT_ID ../nextstrain.org/env/testing/config.json)"
 
     nextstrain login
     nextstrain whoami
     nextstrain remote ls groups/test-private
 
+Most of the times the above is not necessary, however, as you can specify the
+local remote explicitly instead of pretending it's nextstrain.org, e.g.:
```

Comment:
I interpret this section as saying the following are equivalent:

1. Set default remote via env variable, e.g. `NEXTSTRAIN_DOT_ORG=http://localhost:5000 nextstrain login` and `NEXTSTRAIN_DOT_ORG=http://localhost:5000  nextstrain remote ls groups/test`.
2. Specify remote in command, e.g. `nextstrain login localhost:5000`, `nextstrain remote ls http://localhost:5000/groups/test`

And that neither will access nextstrain.org. Is this correct?

Reply:
I have only a cursory understanding of how the CLI, nextstrain.org server (perhaps localhost) and cognito are interacting here. From the commands in my original comment, is whichever cognito configuration the localhost (nextstrain.org) server is using being used by the CLI? In other words, the choice of cognito platform is lifted out of the CLI and instead we just configure the remote we want to interact with?

Reply:
> I interpret this section as saying the following are equivalent: […] And that neither will access https://nextstrain.org/. Is this correct?

They're equivalent except for the short code path before `NEXTSTRAIN_DOT_ORG` is used to potentially override what "nextstrain.org" means (in `nextstrain.cli.remote.parse_remote_path()`). That is, once the remote is parsed, they're equivalent.

Neither will access https://nextstrain.org itself.

> When we login with `nextstrain login localhost:5000` with the nextstrain.org server running at localhost:5000, is this remote the authz server as far as the CLI is concerned? And thus everything to do with cognito (user pools etc) has been lifted out of the CLI and into the remote? And thus any tokens the CLI receives are only ever interacting with the provided remote?

No, the CLI still interacts with the OIDC/OAuth2 authorization server (aka the IdP), which in our case is Cognito but in CDC's case is Entra ID (née Azure AD). It obtains the IdP information from the remote, so it doesn't have to be hardcoded. I do have a rough WIP of docs for this interaction to help folks understand, but it's not up yet.

Reply:
Ok. At least this way the identity provider configuration has been lifted out of the CLI and into the remote.